ci-scripts/check_generated_server.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+# Script to check a generated server has not changed
+
```

Comment:
Should we add a `set -e` as well?

Reply:
https://www.shellcheck.net/

```bash
$ shellcheck myscript
 
Line 7:
DIR=$(dirname $0)
              ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.

Did you mean:
DIR=$(dirname "$0")
 
Line 8:
cd $DIR/..
^-- [SC2164](https://www.shellcheck.net/wiki/SC2164) (warning): Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
   ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.

Did you mean:
cd "$DIR"/.. || exit
 
Line 14:
status=$(git status --porcelain $server_dir)
                                ^-- [SC2086](https://www.shellcheck.net/wiki/SC2086) (info): Double quote to prevent globbing and word splitting.

Did you mean:
status=$(git status --porcelain "$server_dir")
 
Line 15:
if [ $? -ne 0 ]
     ^-- [SC2181](https://www.shellcheck.net/wiki/SC2181) (style): Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.
```

🤷‍♂️

Reply:
Unfortunately no, that would have been simpler. However the `grep -v README.md` call can fail if the README didn't change (which is expected) and then we get a false negative. So we have to handle failures individually.